File: kotlin/fizzbuzz/src/test/kotlin/FizzBuzzTest.kt
import io.kotlintest.specs.FunSpec
import org.assertj.core.api.Assertions.assertThat

class FizzBuzzTest : FunSpec() {
    private var target = Calculator()

    init {
        test("normal numbers should be returned as is", {
            assertThat(target.next(1)).isEqualTo("1")
            assertThat(target.next(2)).isEqualTo("2")
        })

        test("numbers divisible by 3 should return Fizz", {
            assertThat(target.next(3)).isEqualTo("Fizz")
            assertThat(target.next(6)).isEqualTo("Fizz")
        })
        
        test("numbers divisible by 5 should return Buzz", {
            assertThat(target.next(5)).isEqualTo("Buzz")
            assertThat(target.next(10)).isEqualTo("Buzz")
        })
        
        test("numbers divisible by 3 and 5 should return Fizz-Buzz", {
            assertThat(target.next(15)).isEqualTo("Fizz-Buzz")
            assertThat(target.next(30)).isEqualTo("Fizz-Buzz")
        })
    }
}

File: scala/higher-order-functions-demo/src/test/scala/FizzBuzzTest.scala
import org.scalatest.{FlatSpec, Matchers}

class FizzBuzzTest extends FlatSpec with Matchers {
    val generator = new FizzBuzzGenerator()

    behavior of "Fizz-Buzz"

    it should "return normal numbers as is" in {
        generator.nextNumber(1) should be("1")
        generator.nextNumber(2) should be("2")
    }

    it should "return numbers divisible by 3 as Fizz" in {
        generator.nextNumber(3) should be("Fizz")
        generator.nextNumber(6) should be("Fizz")
    }

    it should "return numbers divisible by 5 as Buzz" in {
        generator.nextNumber(5) should be("Buzz")
        generator.nextNumber(10) should be("Buzz")
    }

    it should "return numbers divisible by 3 and 5 as Fizz-Buzz" in {
        generator.nextNumber(15) should be("Fizz-Buzz")
        generator.nextNumber(30) should be("Fizz-Buzz")
    }

}